router/src/server/http/write/single_tenant.rs

```diff
@@ -137,9 +147,12 @@ async fn parse_v1(
             )
         }
     })?;
-    authorize(auth_service, req, &namespace, write_params.p)
-        .await
-        .map_err(SingleTenantExtractError::Authorizer)?;
+
+    let start_instant = Instant::now();
+    let authz_result = authorize(auth_service, req, &namespace, write_params.p).await;
+    authz_result.map_err(SingleTenantExtractError::Authorizer)?;
+    let duration = start_instant.elapsed();
+    duration_histogram_recorder.record(duration);
```

Comment:
I think you want `parse_v1()` and `parse_v2()` to have the same ordering around the call to `map_err()`, as they'll be recording different metrics otherwise. I imagine it would be ideal to track the latency for all requests like the ordering below:

```suggestion  
    let duration = start_instant.elapsed();
    duration_histogram_recorder.record(duration);
    authz_result.map_err(SingleTenantExtractError::Authorizer)?;
```

I suppose the only concern with that is tracking durations for all failed call will fill up the histogram. Maybe @domodwyer has an opinion.

Reply:
Ah, my intent was to do as you suggested -- always record the duration. Just got switched on a rebasing/iteration. Let me make the fix; thanks for catching.

Reply:
actually, removing this commit entirely. Metric will be added separately per Dom's request.